decimal64.go

```diff
@@ -187,20 +194,44 @@ func roundStatus(significand uint64, exp int, targetExp int) discardedDigit {
 	if expDiff > 19 && significand != 0 {
 		return lt5
 	}
-	divisor := powersOf10[expDiff]
-	resizedSig := significand / divisor
-	truncatedSig := significand - resizedSig*divisor
+	remainder := significand % powersOf10[expDiff]
```

Comment:
Suggested rework. It's more efficient, but I haven't confirmed correctness:
```go
switch significand / powersOf10[expDiff - 1] % 10 {
case 0:
	return eq0
case 1, 2, 3, 4:
	return lt 5
case 5:
	return eq5
default:
	return gt5
}
```

Reply:
I think we already touched on this
this method only gives precision to one decimal place; it would work for the comparison of 4<5, but in the case where the discarded digit was one power of ten less than the mid point it would be rounded to zero. 

eg. `50001/10000 == 5 --> eq5` when expected would be gt5

Reply:
What about:
```go
p := powersOf10[expDiff - 1]
switch (significand + p - 1) / p % 10 {
```
(I gave this about 10 seconds of thought, so no idea if it works.)

Reply:
OK, I thought a bit more about it, and no that doesn't work.